website_event_crm/models/crm_lead.py

```diff
@@ -65,6 +48,47 @@ def _cron_auto_invite_website_event_type(self):
             except Exception:
                 _logger.exception("Failure trying to invite to website event type.")
 
+    @api.model
+    def _cron_auto_invite_website_event_type(self):
+        """Invite automatically leads/opportunities to website event categories. We
+        should be aware of multi-company + multi-website scenarios in which we want
+        the link sent to the customer launched only when there are events for their
+        company leads. Moreover, we want to send the proper base url link for the
+        proper filtered events.
+        - crm.leads company isn't mandatory (although unusual)
+        - event.event company isn't mandatory either
+        - website isn't mandatory in either both models
+        """
+        # Find opportunities
+        leads = self.search(
+            [
+                ("email_from", "!=", False),
+                ("event_type_id", "!=", False),
+                ("stage_id.auto_invite_website_event_type", "=", True),
+            ]
+        )
+        available_event_types = leads.mapped("event_type_id")
+        published_events = self.env["event.event"].search(
+            available_event_types._published_events_domain()
+        )
+        # Leads with falsey company won't get invitations
```

Comment:
```suggestion
        # Leads with falsy company won't get invitations
```

Shouldn't this raise an exception if we include any of the ignored ones? If not, people will continue thinking that they have been invited, and wonder why it didn't get. Anyways, I thought that the final criteria is when no website is set.

Reply:
Well, if the cron is the one inviting I don't think so.

> I thought that the final criteria is when no website is set.

But in the event... anyway, as I put in my previous comment, the main problem here is that website isn't mandatory on events.

The issues are two:

- Avoid sending an invitation when no events are available in the category of the event.
- Send the right base_url for each case.

A quick look over the possible values:

|  *required*  | crm.lead | event.event |
|------------|----------|-------------|
| company_id | False    | True        |
| website_id | False    | False       |

|  *default*   | crm.lead | event.event |
|------------|----------|-------------|
| company_id | True     | True        |
| website_id | False    | False       |

We can take for granted that the leads will have the company set. What I think I'll do is to send the notifications by website but also send those from events with no website with the base url of the first website in the company. So in every case the user will be able to check the event list and no invitation will be left.

Reply:
Then put a warning (in the crm.lead I'm interpreting) in the top like other places when a condition for not being sent is met, saying that no mail will be sent due to ...